src/test_workflow/integ_test/integ_test_suite_opensearch.py

```diff
@@ -71,6 +71,43 @@ def __setup_cluster_and_execute_test_config(self, config: str) -> int:
         if "additional-cluster-configs" in self.test_config.integ_test.keys():
             self.additional_cluster_config = self.test_config.integ_test.get("additional-cluster-configs")
             logging.info(f"Additional config found: {self.additional_cluster_config}")
+
+        if "topology" in self.test_config.integ_test.keys():
+            self.topology = self.test_config.integ_test.get("topology")
+            logging.info(f"topology config found: {self.topology}")
+            if(self.topology == "multi-cluster"):
+                if self.additional_cluster_config is None:
+                    self.additional_cluster_config = {"cluster.name": "opensearch1"}
+                self.additional_cluster_config['cluster.name'] = "opensearch1"
+                self.additional_cluster_config['http.port'] = 9200
+                with LocalTestCluster.create(
+                    self.dependency_installer,
+                    os.path.join(self.work_dir, "1"),
+                    self.component.name,
+                    self.additional_cluster_config,
+                    self.bundle_manifest,
+                    security,
+                    config,
+                    self.test_recorder,
+                    9200
+                ) as (endpoint1, port1):
+                    self.additional_cluster_config['cluster.name'] = "opensearch2"
+                    self.additional_cluster_config['http.port'] = 9201
+                    with LocalTestCluster.create(
+                        self.dependency_installer,
+                        os.path.join(self.work_dir, "2"),
+                        self.component.name,
+                        self.additional_cluster_config,
+                        self.bundle_manifest,
+                        security,
+                        config,
+                        self.test_recorder,
+                        9201
+                    ) as (endpoint2, port2):
+                        os.chdir(self.work_dir)
+                        self.pretty_print_message("Running integration tests for " + self.component.name)
+                        cluster_endpoints = [{"endpoint": endpoint1, "port": port1}, {"endpoint": endpoint2, "port": port2}]
+                        return self.multi_execute_integtest_sh(cluster_endpoints, security, config)
         with LocalTestCluster.create(
```

Comment:
Looks like this cluster is still created even if topology is specified?

Reply:
no, the function returns after the tests are completed.

Reply:
Oh I see. It's hard to read. See my refactoring ask below.

Reply:
Made relevant class to address this.